### PR TITLE
ASIM NetworkSession schema parser with its sample data and tester files for Palo Alto Cortex Data Lake

### PR DESCRIPTION
**Pull request by niralishah-crest**
Change(s):

- Added ASIM NetworkSession schema parser for Palo Alto Cortex Data Lake based on CEF data

Reason for Change(s):

- Initial version ASIM NetworkSession Parser

Version Updated:

- Initial version

Testing Completed:

- Yes

Checked that the validations are passing and have addressed any issues that are present:

- Yes

### CODE REVIEW

**Comment by v-atulyadav** [comment]
Hi @niralishah-crest,
Thank you for raising Pull Request with us! We will review the Pull Request and get back to you by shortly. Thanks

**Review by vakohl — CHANGES_REQUESTED** [review]

**Comment by v-atulyadav** [comment]
Hi @niralishah-crest, please check comments above from @vakohl and act accordingly. Thanks

**Comment by v-atulyadav** [comment]
Hi @niralishah-crest,
To follow up appropriately, we would appreciate some indication of the timeframe for these changes. Thanks

**Comment by v-atulyadav** [comment]
Hi @niralishah-crest, please act on @vakohl's comments. Thanks

**Comment by v-atulyadav** [comment]
Hi @niralishah-crest,
We're waiting for an update on this. Thanks

**Comment by v-atulyadav** [comment]
Hi @niralishah-crest,
A few changes were suggested by @vakohl 3 weeks back. Could you please make those changes and update this request accordingly? Thanks

**Comment by v-atulyadav** [comment]
Hi @niralishah-crest, 
We look forward to hearing back from you. Thanks

**Comment by v-atulyadav** [comment]
Hi @niralishah-crest, please update on above pending comments. Thanks

**Comment by v-rusraut** [comment]
Hi @niralishah-crest,
Please work on above comment.
Thanks

**Comment by v-atulyadav** [comment]
We wanted to check on the status of PR #9126. PR is pending for more than 30 days. Please let us know if you need any assistance to review this PR. Per our standard operating procedures if no response is received in the next 7 business days, we will close this PR. Thank you for your cooperation.

**Comment by jayeshprajapaticrest** [comment]
@v-atulyadav 
Sorry for the inconvenience.
We are unable to get chance to fix the suggested changes.
But now we are working on it and will update the PR with the fix of suggested changes by next week. So please do not close this PR.
Thanks.

**Comment by v-atulyadav** [comment]
Noted @jayeshprajapaticrest. Thanks

**Comment by v-atulyadav** [comment]
Noted @niralishah-crest. Thanks

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionPaloAltoCortexDataLake.yaml [hunk 1]
```diff
@@ -0,0 +1,173 @@
+Parser:
+  Title: Network Session ASIM filtering parser for Palo Alto Cortex Data Lake
```
[the remark this answers is not in the document]
Comment (niralishah-crest) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionPaloAltoCortexDataLake.yaml]: Done

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionPaloAltoCortexDataLake.yaml [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+      "1", "Low",
+      "2", "Low",
+      "3", "Low",
+      "4", "Medium",
+      "5", "Medium",
+      "6", "Medium",
+      "7", "High",
+      "8", "High",
+      "9", "High",
+      "10", "High"
```
[the remark this answers is not in the document]
Comment (niralishah-crest) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionPaloAltoCortexDataLake.yaml:65]: We have done this based on mapping given in CommonSecuritLog table documentation as we don't have numeric mapping of severityLevel available in PaloAlto document.
image
https://learn.microsoft.com/en-us/azure/azure-monitor/reference/tables/commonsecuritylog

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionPaloAltoCortexDataLake.yaml [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+      | parse-kv AdditionalExtensions as (PanOSSessionStartTime: string, PanOSDestinationDeviceHost: string, PanOSSourceDeviceHost: string, PanOSDestinationUUID: string, PanOSDestinationLocation: string, PanOSSourceUUID: string, PanOSDestinationDeviceMac: string, PanOsBytes: long, PanOSIsClienttoServer: string, PanOSSourceLocation: string, PanOSSourceDeviceMac: string, PanOSPacketsReceived: long, PanOSPacketsSent: long, PanOSRuleUUID: int) with (pair_delimiter=";", kv_delimiter="=")
+      | invoke _ASIM_ResolveDvcFQDN('DeviceName')
+      | invoke _ASIM_ResolveDstFQDN('PanOSDestinationDeviceHost')
+      | invoke _ASIM_ResolveSrcFQDN('PanOSSourceDeviceHost')
+      | lookup EventResultDvcActionLookup on DeviceAction
+      | lookup EventSeverityLookup on LogSeverity
+      | extend
+          EventStartTime = todatetime(PanOSSessionStartTime),
+          SrcIpAddr = coalesce(DeviceCustomIPv6Address2, SourceIP),
+          DstIpAddr = coalesce(DeviceCustomIPv6Address3, DestinationIP),
```
[the remark this answers is not in the document]
Comment (niralishah-crest) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionPaloAltoCortexDataLake.yaml]: Done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionPaloAltoCortexDataLake.yaml [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+          NetworkSessionId = tostring(FieldDeviceCustomNumber1),
+          NetworkDuration = toint(FieldDeviceCustomNumber3),
+          DstBytes = tolong(ReceivedBytes),
+          SrcBytes = tolong(SentBytes),
+          SrcDomain = coalesce(SourceNTDomain, SrcDomain),
+          DstDomain = coalesce(DestinationNTDomain, DstDomain)
+      | project-rename
+          EventProduct = DeviceProduct,
+          EventVendor = DeviceVendor,
+          DvcIpAddr = Computer,
```
[the remark this answers is not in the document]
Comment (niralishah-crest) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionPaloAltoCortexDataLake.yaml:158]: Not sure as we don't have a live setup available. But in document, Computer field contains IP Addr while DeviceName contains host value.

File: Parsers/ASimNetworkSession/Tests/PaloAlto_CortexDataLake_ASimNetworkSession_SchemaTest.csv [hunk 5]
```diff
@@ -0,0 +1,81 @@
+Result
+"(1) Warning: Missing recommended field [ASimMatchingHostname]"
+"(1) Warning: Missing recommended field [ASimMatchingIpAddr]"
+"(1) Warning: Missing recommended field [EventResultDetails]"
```
[the remark this answers is not in the document]
Comment (niralishah-crest) [reply Parsers/ASimNetworkSession/Tests/PaloAlto_CortexDataLake_ASimNetworkSession_SchemaTest.csv]: Done based on Reason field. Prepared a lookup based on possible values available in document.
https://docs.paloaltonetworks.com/pan-os/11-0/pan-os-admin/monitoring/use-syslog-for-monitoring/syslog-field-descriptions/traffic-log-fields

File: Sample Data/ASIM/CommonSecurityLog_Schema.csv [hunk 6]
```diff
@@ -0,0 +1,161 @@
+ColumnName,ColumnOrdinal,DataType,ColumnType
+TenantId,0,"System.String",string
+TimeGenerated,1,"System.DateTime",datetime
+DeviceVendor,2,"System.String",string
+DeviceProduct,3,"System.String",string
+DeviceVersion,4,"System.String",string
+DeviceEventClassID,5,"System.String",string
+Activity,6,"System.String",string
```
[the remark this answers is not in the document]
Comment (niralishah-crest) [reply Sample Data/ASIM/CommonSecurityLog_Schema.csv]: Mapped with EventOriginalSubType

File: Sample Data/ASIM/PaloAlto_CortexDataLake_ASimNetworkSession_RawLogs.txt [hunk 7]
```diff
@@ -0,0 +1,2 @@
...
...DWANPolicyName= PanOSLinkSwitches= PanOSSDWANCluster= PanOSSDWANDeviceType= PanOSSDWANClusterType= PanOSSDWANSite= PanOSDynamicUserGroupName=dynug-4 PanOSX-Forwarded-ForIP=198.51.100.0 PanOSSourceDeviceCategory=N-Phone PanOSSourceDeviceProfile=n-profile PanOSSourceDeviceModel=Nexus PanOSSourceDeviceVendor=Google PanOSSourceDeviceOSFamily=LG-H790 PanOSSourceDeviceOSVersion=Android v6 PanOSSourceDeviceHost=pan-301 PanOSSourceDeviceMac=839147449905 PanOSDestinationDeviceCategory=N-Phone PanOSDestinationDeviceProfile=n-profile PanOSDestinationDeviceModel=Nexus PanOSDestinationDeviceVendor=Google PanOSDestinationDeviceOSFamily=H1511 PanOSDestinationDeviceOSVersion=Android v7 PanOSDestinationDeviceHost=pan-355 PanOSDestinationDeviceMac=530589561221 PanOSContainerID=1873cc5c-0d31 PanOSContainerNameSpace=pns_default PanOSContainerName=pan-dp-77754f4 PanOSSourceEDL= PanOSDestinationEDL= PanOSGPHostID=xxxxxxxxxxxxxx PanOSEndpointSerialNumber=xxxxxxxxxxxxxx PanOSSourceDynamicAddressGroup= aqua_dag PanOSDestinationDynamicAddressGroup= PanOSHASessionOwner=session_owner-4 PanOSTimeGeneratedHighResolution=Feb 27 2021 20:16:18 PanOSNSSAINetworkSliceType=0 PanOSNSSAINetworkSliceDifferentiator=1bca5
```
[the remark this answers is not in the document]
Comment (niralishah-crest) [reply Sample Data/ASIM/PaloAlto_CortexDataLake_ASimNetworkSession_RawLogs.txt:2]: Done

File: Parsers/ASimNetworkSession/Tests/PaloAlto_CortexDataLake_ASimNetworkSession_SchemaTest.csv [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+"(2) Info: Missing optional field [SrcGeoLongitude]"
+"(2) Info: Missing optional field [SrcGeoRegion]"
+"(2) Info: Missing optional field [SrcInterfaceGuid]"
+"(2) Info: Missing optional field [SrcInterfaceName]"
+"(2) Info: Missing optional field [SrcOriginalUserType]"
+"(2) Info: Missing optional field [SrcProcessGuid]"
+"(2) Info: Missing optional field [SrcProcessId]"
+"(2) Info: Missing optional field [SrcProcessName]"
+"(2) Info: Missing optional field [SrcScopeId]"
+"(2) Info: Missing optional field [SrcUserId]"
```
[the remark this answers is not in the document]
Comment (niralishah-crest) [reply Parsers/ASimNetworkSession/Tests/PaloAlto_CortexDataLake_ASimNetworkSession_SchemaTest.csv]: Done

File: Sample Data/ASIM/CommonSecurityLog_Schema.csv [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+DeviceCustomFloatingPoint3,104,"System.Double",real
+DeviceCustomFloatingPoint3Label,105,"System.String",string
+DeviceCustomFloatingPoint4,106,"System.Double",real
+DeviceCustomFloatingPoint4Label,107,"System.String",string
+DeviceCustomNumber1,108,"System.Int32",int
+FieldDeviceCustomNumber1,109,"System.Int64",long
+DeviceCustomNumber1Label,110,"System.String",string
+DeviceCustomNumber2,111,"System.Int32",int
+FieldDeviceCustomNumber2,112,"System.Int64",long
+DeviceCustomNumber2Label,113,"System.String",string
```
[the remark this answers is not in the document]
Comment (niralishah-crest) [reply Sample Data/ASIM/CommonSecurityLog_Schema.csv]: NetworkPackets is already mapped with FieldDeviceCustomNumber2

File: Sample Data/ASIM/CommonSecurityLog_Schema.csv [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+DeviceCustomNumber3,114,"System.Int32",int
+FieldDeviceCustomNumber3,115,"System.Int64",long
+DeviceCustomNumber3Label,116,"System.String",string
```
[the remark this answers is not in the document]
Comment (niralishah-crest) [reply Sample Data/ASIM/CommonSecurityLog_Schema.csv]: NetworkDuration is already mapped with FieldDeviceCustomNumber3

File: Sample Data/ASIM/CommonSecurityLog_Schema.csv [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+DeviceCustomString1,117,"System.String",string
+DeviceCustomString1Label,118,"System.String",string
```
[the remark this answers is not in the document]
Comment (niralishah-crest) [reply Sample Data/ASIM/CommonSecurityLog_Schema.csv]: NetworkRuleName is already mapped with DeviceCustomString1

File: Sample Data/ASIM/CommonSecurityLog_Schema.csv [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+DeviceCustomString2,119,"System.String",string
+DeviceCustomString2Label,120,"System.String",string
+DeviceCustomString3,121,"System.String",string
+DeviceCustomString3Label,122,"System.String",string
+DeviceCustomString4,123,"System.String",string
+DeviceCustomString4Label,124,"System.String",string
+DeviceCustomString5,125,"System.String",string
+DeviceCustomString5Label,126,"System.String",string
```
[the remark this answers is not in the document]
Comment (niralishah-crest) [reply Sample Data/ASIM/CommonSecurityLog_Schema.csv]: Done.
SrcZone = DeviceCustomString4
DstZone = DeviceCustomString5

File: Sample Data/ASIM/CommonSecurityLog_Schema.csv [hunk 13]
```diff
@@ -0,0 +1,161 @@
...
+IndicatorThreatType,147,"System.String",string
+ThreatDescription,148,"System.String",string
+ThreatConfidence,149,"System.String",string
+ReportReferenceLink,150,"System.String",string
+MaliciousIPLongitude,151,"System.Double",real
+MaliciousIPLatitude,152,"System.Double",real
+MaliciousIPCountry,153,"System.String",string
+Computer,154,"System.String",string
+SourceSystem,155,"System.String",string
+SimplifiedDeviceAction,156,"System.String",string
```
[the remark this answers is not in the document]
Comment (niralishah-crest) [reply Sample Data/ASIM/CommonSecurityLog_Schema.csv]: SimplifiedDeviceAction contains value in simplifiedform for DeviceAction. e.g. if DeviceAction field contains value as "Allowed" then SimplifiedDeviceAction field will have "Allow" as value.

**Comment by niralishah-crest** [comment]
@v-atulyadav @vakohl 
We have addressed all review comments.
Thanks

**Comment by v-atulyadav** [comment]
Thanks @niralishah-crest.

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionPaloAltoCortexDataLake.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (vakohl) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionPaloAltoCortexDataLake.yaml:65]: @niralishah-crest @jayeshprajapaticrest 
We should not follow generic metric for this.

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionPaloAltoCortexDataLake.yaml [hunk 14]
```diff
@@ -0,0 +1,173 @@
...
+          User = DstUsername,
+          Hostname = DstHostname,
+          SrcDvcIdType = iff(isnotempty(SrcDvcId), "Other", ""),
+          DstDvcIdType = iff(isnotempty(DstDvcId), "Other", ""),
+          SrcDomainType = iff(isnotempty(SourceNTDomain), "Windows", SrcDomainType),
+          DstDomainType = iff(isnotempty(DestinationNTDomain), "Windows", DstDomainType)
+      | extend
+          EventSchema = "NetworkSession",
+          EventSchemaVersion = "0.2.6",
+          EventType = "NetworkSession"
```
[the remark this answers is not in the document]
Comment (vakohl) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionPaloAltoCortexDataLake.yaml]: @niralishah-crest @jayeshprajapaticrest 
pls update

File: Parsers/ASimNetworkSession/Tests/PaloAlto_CortexDataLake_ASimNetworkSession_SchemaTest.csv (hunk 5 above)
[the remark this answers is not in the document]
Comment (vakohl) [reply Parsers/ASimNetworkSession/Tests/PaloAlto_CortexDataLake_ASimNetworkSession_SchemaTest.csv]: great

File: Sample Data/ASIM/PaloAlto_CortexDataLake_ASimNetworkSession_RawLogs.txt (hunk 7 above)
[the remark this answers is not in the document]
Comment (vakohl) [reply Sample Data/ASIM/PaloAlto_CortexDataLake_ASimNetworkSession_RawLogs.txt:2]: @niralishah-crest could find it mapped. Please re-check

File: Sample Data/ASIM/PaloAlto_CortexDataLake_ASimNetworkSession_RawLogs.txt [hunk 15]
```diff
@@ -0,0 +1,2 @@
...
```
[the remark this answers is not in the document]
Comment (vakohl) [reply Sample Data/ASIM/PaloAlto_CortexDataLake_ASimNetworkSession_RawLogs.txt:1]: @niralishah-crest is this done?

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionPaloAltoCortexDataLake.yaml [hunk 16]
```diff
@@ -0,0 +1,173 @@
...
+      | extend
+          EventStartTime = todatetime(PanOSSessionStartTime),
+          SrcIpAddr = coalesce(DeviceCustomIPv6Address2, SourceIP),
+          DstIpAddr = coalesce(DeviceCustomIPv6Address3, DestinationIP),
+          NetworkSessionId = tostring(FieldDeviceCustomNumber1),
+          NetworkDuration = toint(FieldDeviceCustomNumber3),
+          DstBytes = tolong(ReceivedBytes),
+          SrcBytes = tolong(SentBytes),
+          SrcDomain = coalesce(SourceNTDomain, SrcDomain),
+          DstDomain = coalesce(DestinationNTDomain, DstDomain)
```
[the remark this answers is not in the document]
Comment (vakohl) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionPaloAltoCortexDataLake.yaml]: @niralishah-crest 
please check this

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionPaloAltoCortexDataLake.yaml [hunk 17]
```diff
@@ -0,0 +1,362 @@
...
+          IpAddr = SrcIpAddr,
+          SessionId = NetworkSessionId,
+          User = DstUsername,
+          Hostname = DstHostname,
+          SrcDvcIdType = iff(isnotempty(SrcDvcId), "Other", ""),
+          DstDvcIdType = iff(isnotempty(DstDvcId), "Other", ""),
+          SrcDomainType = iff(isnotempty(SourceNTDomain), "Windows", SrcDomainType),
+          DstDomainType = iff(isnotempty(DestinationNTDomain), "Windows", DstDomainType),
+          SrcUserIdType = iff(isnotempty(SrcUserId), "UID", ""),
+          DstUserIdType = iff(isnotempty(DstUserId), "UID", ""),
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionPaloAltoCortexDataLake.yaml:305]: @niralishah-crest 
is this linux user id?
image

Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionPaloAltoCortexDataLake.yaml:305]: @niralishah-crest 
are values of linux user id type?
image

File: Sample Data/ASIM/CommonSecurityLog_Schema.csv (hunk 9 above)
[the remark this answers is not in the document]
Comment (vakohl) [reply Sample Data/ASIM/CommonSecurityLog_Schema.csv]: ok

File: Sample Data/ASIM/CommonSecurityLog_Schema.csv (hunk 10 above)
[the remark this answers is not in the document]
Comment (vakohl) [reply Sample Data/ASIM/CommonSecurityLog_Schema.csv]: ok

File: Sample Data/ASIM/CommonSecurityLog_Schema.csv (hunk 11 above)
[the remark this answers is not in the document]
Comment (vakohl) [reply Sample Data/ASIM/CommonSecurityLog_Schema.csv]: ok

File: Sample Data/ASIM/CommonSecurityLog_Schema.csv (hunk 13 above)
[the remark this answers is not in the document]
Comment (vakohl) [reply Sample Data/ASIM/CommonSecurityLog_Schema.csv]: ok. DvcAction is mapped already

File: Parsers/ASimNetworkSession/Tests/PaloAlto_CortexDataLake_ASimNetworkSession_DataTest.csv [hunk 18]
```diff
@@ -0,0 +1,7 @@
+Result
+"(0) Error: 2 invalid value(s) (up to 10 listed) in 6 records (200.0%) for field [EventVendor] of type [Enumerated]: [""Palo Alto Networks""] (Schema:NetworkSession)"
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Tests/PaloAlto_CortexDataLake_ASimNetworkSession_DataTest.csv:2]: @niralishah-crest 
duplicate entry. Palo Alto is already present. Please remove this change

**Comment by v-atulyadav** [comment]
Hi @niralishah-crest,
There are some changes that Varun has suggested, please take a look at them. Thanks

**Comment by v-atulyadav** [comment]
Hi @niralishah-crest,
We're waiting for an update on this. Thanks

File: ASIM/dev/ASimTester/ASimTester.csv [hunk 19]
```diff
@@ -679,7 +679,7 @@ EventVendor,string,Mandatory,Authentication,Enumerated,Salesforce|AWS|Barracuda|
 EventVendor,string,Mandatory,Common,,,
 EventVendor,string,Mandatory,Dhcp,,,
 EventVendor,string,Mandatory,Dns,Enumerated,Cisco|Corelight|GCP|Infoblox|Microsoft|Zscaler|Vectra AI|SentinelOne,
-EventVendor,string,Mandatory,FileEvent,Enumerated,Microsoft|SentinelOne,
+EventVendor,string,Mandatory,FileEvent,Enumerated,Microsoft|SentinelOne|Palo Alto Networks,
```
Comment (vakohl) [inline ASIM/dev/ASimTester/ASimTester.csv:682]: @niralishah-crest @jayeshprajapaticrest 
why updating FileEvent entry?

File: ASIM/dev/ASimTester/ASimTester.csv [hunk 20]
```diff
@@ -544,7 +544,7 @@ EventProduct,string,Mandatory,FileEvent,Enumerated,Sysmon for Linux|Sysmon|M365
...
 EventProduct,string,Mandatory,FileEvent,Enumerated,Sysmon for Linux|Sysmon|M365 Defender for Endpoint|Azure File Storage|SharePoint|OneDrive|SentinelOne,
 EventProduct,string,Mandatory,NetworkSession,Enumerated,Fortigate|IOS|ISE|SDP|Vectra Stream|NSGFlow|Fireware|VPC|Azure Defender for IoT|Azure Firewall|M365 Defender for Endpoint|Sysmon|Sysmon for Linux|Windows Firewall|WireData|ZIA Firewall|CDL|PanOS|VMConnection|Meraki|Zeek|Firewall|ASA|Cynerio|SentinelOne|WAF|Firepower|FalconHost,
-EventProduct,string,Mandatory,NetworkSession,Enumerated,Fortigate|IOS|ISE|SDP|Vectra Stream|NSGFlow|Fireware|VPC|Azure Defender for IoT|Azure Firewall|M365 Defender for Endpoint|Sysmon|Sysmon for Linux|Windows Firewall|WireData|ZIA Firewall|CDL|PanOS|VMConnection|Meraki|Zeek|Firewall|ASA|Cynerio|SentinelOne|WAF|Firepower|Carbon Black Cloud,
+EventProduct,string,Mandatory,NetworkSession,Enumerated,Fortigate|IOS|ISE|SDP|Vectra Stream|NSGFlow|Fireware|VPC|Azure Defender for IoT|Azure Firewall|M365 Defender for Endpoint|Sysmon|Sysmon for Linux|Windows Firewall|WireData|ZIA Firewall|CDL|PanOS|VMConnection|Meraki|Zeek|Firewall|ASA|Cynerio|SentinelOne|WAF|Firepower|Carbon Black Cloud| Cortex Data Lake,
```
Comment (vakohl) [inline ASIM/dev/ASimTester/ASimTester.csv:547]: @niralishah-crest seems like an extra space

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionPaloAltoCortexDataLake.yaml [hunk 21]
```diff
@@ -0,0 +1,362 @@
...
+      ) {
+      let src_or_any = set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+      let dst_or_any = set_union(dstipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+      CommonSecurityLog
+      | where not(disabled)
+          and (isnull(starttime) or TimeGenerated >= starttime)
+          and (isnull(endtime) or TimeGenerated <= endtime)
+          and DeviceVendor == "Palo Alto Networks" and DeviceProduct == "LF"
+          and DeviceEventClassID == "TRAFFIC"
+          and (eventresult == "*" or eventresult == "Success")
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionPaloAltoCortexDataLake.yaml:125]: @niralishah-crest @jayeshprajapaticrest 
Don't think this is correct, we are setting both success and failure. please validate

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionPaloAltoCortexDataLake.yaml [hunk 22]
```diff
@@ -0,0 +1,362 @@
...
+          and (array_length(hostname_has_any) == 0 or AdditionalExtensions has_any (hostname_has_any))
+          and (isnull(dstportnumber) or toint(DestinationPort) == dstportnumber)
+          and (array_length(dvcaction) == 0)
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionPaloAltoCortexDataLake.yaml:128]: @niralishah-crest @jayeshprajapaticrest 
DeviceAction value being set. Please validate this filter

**Comment by v-atulyadav** [comment]
Hi @niralishah-crest, please check comments above from @vakohl and act accordingly. Thanks

**Comment by v-atulyadav** [comment]
Hi @niralishah-crest, please act on above comments. Also check for branch conflicts. Thanks

**Comment by v-atulyadav** [comment]
Hi @niralishah-crest,
There are some changes that Varun has suggested, please take a look at them. Thanks

**Comment by v-atulyadav** [comment]
Hi @niralishah-crest,
Please work on above comment. Also look for a branch conflicts. Thanks

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionPaloAltoCortexDataLake.yaml (hunk 14 above)
[the remark this answers is not in the document]
Comment (vakohl) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionPaloAltoCortexDataLake.yaml]: NetworkSession seems valid.

File: Parsers/ASimNetworkSession/Tests/PaloAlto_CortexDataLake_ASimNetworkSession_SchemaTest.csv (hunk 5 above)
[the remark this answers is not in the document]
Comment (vakohl) [reply Parsers/ASimNetworkSession/Tests/PaloAlto_CortexDataLake_ASimNetworkSession_SchemaTest.csv]: couldn't find metric but sample log below has reason field.
https://docs.paloaltonetworks.com/cortex/cortex-data-lake/log-forwarding-schema-reference/network-logs/network-traffic-log/network-traffic-cef-fields

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionPaloAltoCortexDataLake.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (vakohl) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionPaloAltoCortexDataLake.yaml:65]: keeping 9 and 10 as high only.

File: Sample Data/ASIM/PaloAlto_CortexDataLake_ASimNetworkSession_RawLogs.txt (hunk 15 above)
[the remark this answers is not in the document]
Comment (vakohl) [reply Sample Data/ASIM/PaloAlto_CortexDataLake_ASimNetworkSession_RawLogs.txt:1]: no response, closing comment.

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionPaloAltoCortexDataLake.yaml (hunk 16 above)
[the remark this answers is not in the document]
Comment (vakohl) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionPaloAltoCortexDataLake.yaml]: based on sample log. Couldn't find this value in sample log.

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionPaloAltoCortexDataLake.yaml (hunk 17 above)
Comment (vakohl) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionPaloAltoCortexDataLake.yaml:305]: userid mapping seems correct as per documentation - https://docs.paloaltonetworks.com/cortex/cortex-data-lake/log-forwarding-schema-reference/network-logs/network-traffic-log#network-traffic_network-traffic-source-user-info.uuid
but could validate id type as sample value not available.

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionPaloAltoCortexDataLake.yaml (hunk 21 above)
Comment (vakohl) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionPaloAltoCortexDataLake.yaml:125]: I've updated filter to check with EventResult field

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionPaloAltoCortexDataLake.yaml (hunk 22 above)
Comment (vakohl) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionPaloAltoCortexDataLake.yaml:128]: I've corrected this.

File: ASIM/dev/ASimTester/ASimTester.csv (hunk 20 above)
Comment (vakohl) [reply ASIM/dev/ASimTester/ASimTester.csv:547]: fixed

File: ASIM/dev/ASimTester/ASimTester.csv (hunk 19 above)
Comment (vakohl) [reply ASIM/dev/ASimTester/ASimTester.csv:682]: removed

File: Parsers/ASimNetworkSession/Tests/PaloAlto_CortexDataLake_ASimNetworkSession_DataTest.csv (hunk 18 above)
Comment (vakohl) [reply Parsers/ASimNetworkSession/Tests/PaloAlto_CortexDataLake_ASimNetworkSession_DataTest.csv:2]: fixed in parser

**Review by vakohl — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**